src/extendables/User/Bank.ts

```diff
@@ -159,31 +159,33 @@ export default class extends Extendable {
 	}
 
 	public async removeItemsFromBank(this: User, _itemBank: O.Readonly<ItemBank>) {
-		const itemBank = _itemBank instanceof Bank ? { ..._itemBank.bank } : _itemBank;
-
-		await this.settings.sync(true);
+		return this.queueFn(async user => {
```

Comment:
You can use `this` within the function body, so not sure why you are changing the prototypes, calling convention for queue.add(), and passing user in as a variable. Let me know if i'm missing something, please :)